src/build/index.ts

```diff
@@ -381,6 +384,24 @@ export default class StreamsBuild {
         }
     }
 
+    private static async checkIfDirty(filePath): Promise<void> {
+        let isFileDirty = false;
+        let dirtyFile = null;
+        const fileName = filePath.substring(filePath.lastIndexOf("/") + 1);
+        for (let i = 0; i < workspace.textDocuments.length; i += 1) {
+            if (workspace.textDocuments[i].fileName === filePath && workspace.textDocuments[i].isDirty) {
+                isFileDirty = true;
+                dirtyFile = workspace.textDocuments[i];
+            }
+        }
```

Comment:
Try using `.find` instead of a `for` loop to make this cleaner (and since there will only be one matching file). Then, we avoid the need to have the `isFileDirty` and `dirtyFile` temp variables.
```
const dirtyFile = workspace.textDocuments.find(...);
```